numba/targets/arraymath.py

```diff
@@ -4218,3 +4218,39 @@ def impl(a, b):
         return _cross2d_operation(a_, b_)
 
     return impl
+
+# ----------------------------------------------------------------------------
+# Set routines
+
+
+@overload(np.setxor1d)
+def np_setxor1d(ar1, ar2, assume_unique=False):
+    if not (type_can_asarray(ar1) and type_can_asarray(ar2)):
+        raise TypingError("Inputs must be array-like")
+
+    if not isinstance(assume_unique, types.Boolean):
+        raise TypingError("Argument `assume_unique` must be boolean")
+
+    if ar1.ndim != 1 or ar2.ndim != 1:
+        raise TypingError("Only 1D arrays input arrays are supported")
+
+    def impl(ar1, ar2, assume_unique=False):
+        a = np.asarray(ar1)
+        b = np.asarray(ar2)
+
+        if not assume_unique:
+            a = np.unique(a)
+            b = np.unique(b)
```

Comment:
I think it's this branch that's causing the type unification problem if the inputs are `nd`. One way around this would be to create an else branch that yields variables of the same type on exit of the block. For example:
```python
        if not assume_unique:
            au = np.unique(a)
            bu = np.unique(b)
        else:
            au = a.ravel()
            bu = b.ravel()

        aux = np.concatenate((au, bu))
```
in the above `au`, and `bu` will end up as 1d arrays by the time they reach the `concatenate`.

This is a deviation from NumPy's algorithm but I don't think it has any impact to the logic? If `assume_unique` is `False` then `np.unique` will run on the input arrays to unique them and return a 1D array, if `assume_unique` is `True` then the input array's are already solely unique values (and assumed ordered!?!?) and `ravel`ing them will be ok?